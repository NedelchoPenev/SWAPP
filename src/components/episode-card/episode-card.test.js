import React from 'react';
import { shallow } from 'enzyme';

import EpisodeCard from './episode-card.component';

describe('EpisodeCard component', () => {
  let wrapper;
  const mockId = 'episode.1';
  const imageUrl = 'www.testImage.com';
  const mockTitle = 'The Phantom Menace';
  const mockOpeningCrawl = 'Test test test';
  let mockHistory;
  let mockMatch;
  let mockProps;

  beforeEach(() => {
    mockHistory = { push: jest.fn() };
    mockMatch = {
      path: '/episodes',
    };

    mockProps = {
      epsisode: {
        id: mockId,
        title: mockTitle,
        openingCrawl: mockOpeningCrawl,
        image: imageUrl,
      },
      history: mockHistory,
      match: mockMatch,
    };

    wrapper = shallow(<EpisodeCard.WrappedComponent {...mockProps} />);
  });

  it('should render EpisodeCard component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call history.push with the right string when EpisodeCard clicked', () => {
    wrapper.simulate('click');

    expect(mockHistory.push).toHaveBeenCalledWith(
      `${mockMatch.path}/${mockId}`,
    );
  });

  it('should render imageUrl', () => {
    expect(wrapper.find('img').prop('src')).toBe(imageUrl);
  });

  it('should render title', () => {
    expect(wrapper.find('Title').text()).toBe(mockTitle);
  });

  it('should render description', () => {
    expect(wrapper.find('Description').text()).toBe(mockOpeningCrawl);
  });
});
