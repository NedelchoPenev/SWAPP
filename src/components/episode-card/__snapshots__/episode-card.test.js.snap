// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeCard component should render EpisodeCard component 1`] = `
<styled.div
  onClick={[Function]}
>
  <styled.span>
    <styled.span>
      <img
        alt="The Phantom Menace"
        src="www.testImage.com"
      />
    </styled.span>
    <styled.span>
      <Title>
        The Phantom Menace
      </Title>
      <Description>
        Test test test
      </Description>
    </styled.span>
  </styled.span>
</styled.div>
`;
